tutor/src/screens/my-courses/dashboard/index.tsx

```diff
@@ -0,0 +1,194 @@
+    <>
+        {map(courses, c => (<ViewCourse course={c} key={c.id}/>))}
+        <CoursePreview offeringWithCourses={offeringWithCourses} />
+        <CreateACourse />
+    </>
+)
+
+/**
+ * Component that holds the past, current and future courses. Also the resources for the course.
+ */
+const OfferingBlock = ({ offeringWithCourses, isFirstBlock }: {offeringWithCourses: OfferingWithCourses, isFirstBlock: boolean}) => {
+    const [tabIndex, setTabIndex] = useState(0);
+
+    const currentCourses = useMemo(() => filter(offeringWithCourses.courses, c => isCourseCurrent(c), offeringWithCourses))
+    const pastCourses = useMemo(() => filter(offeringWithCourses.courses, c => isCoursePast(c), offeringWithCourses))
+
+    const showTabInfo = useCallback(() => {
+        switch(tabIndex) { 
+            case 0: { 
+                return <CurrentCourses courses={currentCourses} offeringWithCourses={offeringWithCourses} />
+            } 
+            case 1: { 
+                return <PastCourses courses={pastCourses} />
+            } 
+            case 2: { 
+                return <ResourcesInfo appearanceCode={offeringWithCourses.appearance_code} isFirstBlock={isFirstBlock} />
+            } 
+            default: { 
+               return <p>How did you get here?!</p>
+            } 
+         } 
+    }, [tabIndex])
+
+    return (
+        <div className="offering-container">    
+            <h2>{offeringWithCourses.title}</h2>
+            <Tabs
+              tabs={['CURRENT', 'PAST', 'RESOURCES']}
+              onSelect={(a) => setTabIndex(a)}/>
+              <div className="course-cards">
+                {showTabInfo()}
+              </div>
+        </div>
+    )
+}
+
+/**
+ * Main component
+ */
+export const MyCoursesDashboard = ({ offeringsWithCourses }: MyCoursesDashboardProps) => {
+    return (
+        <StyledMyCoursesDashboard>
+            <h2>My Courses</h2>
+            <div className="controls">
+                <Button variant="link"><Icon type="cog" />Manage subjects</Button>
+            </div>
+            { map(offeringsWithCourses, (o, i) => <OfferingBlock key={o.id} offeringWithCourses={o} isFirstBlock={i === 0} /> )}
+        </StyledMyCoursesDashboard>
+    )
+}
+
+const mapStateToProps = (state) => {
+    let offeringsWithCourses: OfferingWithCourses[] = []
+    const courses = state.courses.entities
+    const offerings = state.offerings.entities
+
+    if(!isEmpty(courses) && !isEmpty(offerings)) {
+        offeringsWithCourses = map(offerings, o => {
+            const offeringCourses = compact(
+                    map(courses, c => {
+                if(c.offering_id === o.id) return c;
+                return null
+            }));
+            return { ...o, courses: offeringCourses }
+        });
+    }
+
+    return {
+        offeringsWithCourses,
+    }
+}
+
+export default connect(mapStateToProps)(MyCoursesDashboard)
```

Comment:
a few more suggestions.  

Most of this probably belongs inside a new file for the teacher-has-courses case.  Maybe `screens/my-courses/courses` ?  That would leave index.ts only containing the logic for switching between the create course and display course screens.

I think instead of using `connect` here we can use hooks almost all the time, either with redux `useStore` or use the exported selectors.  For instance to get all the courses inside a component you could:
```ts
import { selectors  }  from 'store/selectors'
// ... inside component
const courses  = selectors.selectAll()
```

Reply:
I have all of these files for the "teacher-has-courses" case inside of `screens/my-courses/dashboard`. 

Yes, I am switching to it now. FTW hooks now. :)